iphone/Maps/UI/PlacePage/PlacePageLayout/ActionBar/MWMActionBarButton.m

```diff
@@ -182,4 +190,14 @@ - (void)setupBookmarkButton:(BOOL)isSelected {
   animationIV.animationRepeatCount = 1;
 }
 
+- (BOOL)needsToHighlightRouteToButton {
+  NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+  if ((FirstSession.isFirstSession || FirstSession.isFirstSessionInCurrentVersion)
+    && ![userDefaults boolForKey:kUDDidHighlightRouteToButton]) {
+    [userDefaults setBool:true forKey:kUDDidHighlightRouteToButton];
```

Comment:
Is this approach reliable when PP is recreated on a second click?

A more robust solution is to highlight it until it is not pressed the first time.

Reply:
Refactor this to:
`return  ![NSUserDefaults.standardUserDefaults boolForKey:kUDDidHighlightRouteToButton] ;`
So it works the same manner but without FirstSession